Tests/ApolloTests/GraphQLExecutor_SelectionSetMapper_FromResponse_Tests.swift

```diff
@@ -1414,4 +1466,53 @@ class GraphQLExecutor_SelectionSetMapper_FromResponse_Tests: XCTestCase {
     expect(data.name).to(beNil())
   }
 
+  func test__booleanCondition_bothSkipAndInclude_onSeperateFieldsForSameSelectionMergedAsOrCondition__givenSkipIsTrue_includeIsFalse_doesNotGetValuesForConditionalFields() throws {
+    // given
+    class GivenSelectionSet: MockSelectionSet {
+      override class var selections: [Selection] {[
+        .include(if: "include" || !"skip", .field("name", String.self))
+      ]}
+    }
+    let object: JSONObject = ["name": "Luke Skywalker", "id": "1234"]
+    let variables = ["skip": true,
+                     "include": false]
+
+    // when
+    let data = try readValues(GivenSelectionSet.self, from: object, variables: variables)
+
+    // then
+    expect(data.name).to(beNil())
+  }
+
+  func test__booleanCondition_bothSkipAndInclude_mergedAsComplexLogicalCondition_correctlyEvaluatesConditionalSelections() throws {
+    // given
+    class GivenSelectionSet: MockSelectionSet {
+      override class var selections: [Selection] {[
+        .include(if: ("a" && !"b" && "c") || "d" || !"e", .field("name", String.self))
```

Comment:
The parenthesis confused me here but that's not how this condition is being evaluated. Did you intend to have grouping of the conditions in the parenthesis; variables below would have to be grouped as arrays for it to function like that?

Depending on your intent for this test I think we should either remove the parenthesis here or create another test that evaluates both && and || groups.